test/blackbox-tests/test-cases/directory-targets/main.t

```diff
@@ -31,8 +31,7 @@ Directory targets are not allowed for non-sandboxed rules.
   1 | (rule
   2 |   (targets (dir output))
   3 |   (action (bash "true")))
-  Error: Rules with directory targets must be sandboxed.
-  Hint: Add (sandbox always) to the (deps ) field of the rule.
+  Error: Rule failed to produce directory "output"
```

Comment:
This test is now identical to the one below, so we should just drop it.

Reply:
Oh, sorry, never mind. It's not identical! Please adjust the test header.

Reply:
Actually, perhaps we should make a copy of this test, to test the same set of properties but without sandboxing?

Reply:
Will do. It's a shame that something like this is needed though. It's not the first time I had to do this copy pasting of tests.

Reply:
Actually, I just looked and many of the tests in `main.t` wouldn't benefit from removing sandboxing. E.g. requesting directory targets from the command line has nothing to do with sandboxing so it's pointless to duplicate the test. I'll extract only the useful tests.

Reply:
Sure, it would be great it we could test everything we care about without duplicating every test.

Having said that, I don't see copying tests as a big problem. I find that some amount of duplication with docs and tests is common and sometimes unavoidable, e.g., if there are 4 yes/no flags affecting the behaviour of a program in an orthogonal way, you've got to test all 2^4 combinations!

Reply:
Indeed, it's useful if it's testing new code paths. But here I think it's basically just testing the same thing. Have a look at the updated test files. Let me know if you think there's anything missing.